### PR TITLE
make ingress-inquisition display information about the relevant security groups and vpcs

### PR DESCRIPTION
**Pull request by NovemberTang**
## How to review

This PR will be easiest to understand if digested one commit at a time. Luckily, there's only 3 commits. Most of it is lockfile changes, and code moving around.

## What does this change?

- Renamed `aws-auth` -> `aws-common`, as it now contains more than just auth logic
- Moved `ReturnFindings` into `aws-common`, as it's used in both tools
- In `ingress-inquisition`, take the list of VPCs failing EC2.2, and return some information about the VPCs that they belong to
- create utils file for bits of logic that were cluttering up `main`

## How to test

```(cd ingress-inquisition && go run main.go --profile <PROFILE> --region <REGION>)```

Using the brackets will return you to your original directory after execution

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
 
       - name: Run tests
-        run: go test ./bucket-blocker/...
+        run: go list -f '{{.Dir}}/...' -m | xargs go test
```
Comment (NovemberTang) [inline .github/workflows/ci.yaml:28]: This tests every module. Go doesn't have great support for running the same command across multiple workspaces as far as i can see

File: aws-common/auth.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	stsClient := sts.NewFromConfig(cfg)
+	_, err = validateCredentials(ctx, stsClient, profile)
+	if err != nil {
+		return cfg, err
+	}
+
+	return cfg, nil
+}
+
+func ReturnFindings(ctx context.Context, securityHubClient *securityhub.Client, controlId string, maxResults int32) (*securityhub.GetFindingsOutput, error) {
```
Comment (NovemberTang) [inline aws-common/auth.go:39]: This is the only new function. The rest was just a migration from `aws-auth` -> `aws-common`

File: ingress-inquisition/go.mod
Comment (NovemberTang) [inline ingress-inquisition/go.mod:1]: The diff here looks extremely weird in unified view. It makes more sense in split view

File: ingress-inquisition/main.go
Comment (NovemberTang) [inline ingress-inquisition/main.go:1]: The diff here looks extremely weird as a lot is new, and a lot has been deleted. You're better off just acting as if this is a brand new file, and ignoring the diff completely

**Review by tjsilver — APPROVED** [review]
Tested locally and works as expected! 👍

**Pull request merged**